### DISCUSSION TITLE
How can I invoke a method that uses an enum for its parameter?

### DISCUSSION THREAD

**Discussion by RealIndica** [vfsfitvnm/frida-il2cpp-bridge#126] [Q&A]
I am messing around with a game and trying to call a function which needs an enum as a parameter.

Dnspy :
```cs
  public void OverrideInteractionGradingType(GameInteractionData.InteractionGradingType gradingType)
  {
  }
		
  public enum InteractionGradingType
  {
      Normal,
      AllFlawless
  }
```

My code looks like so :
```ts
  enum InteractionGradingType
  {
      Normal,
      AllFlawless
  }

  function OverrideInteractionGradingType(interactionGradingType: InteractionGradingType){
      if (isGameActive){
          gameControllerInstance.methods.OverrideInteractionGradingType.invoke(interactionGradingType as number);
      }
  }
```
  
  And throws the error : 
```css
  TypeError: cannot read property 'message' of null
    at try (node_modules/frida-il2cpp-bridge/dist/il2cpp/base.js:119)
    at invokeRaw (node_modules/frida-il2cpp-bridge/dist/il2cpp/structs/method.js:175)
    at OverrideInteractionGradingType (agent/index.ts:110)
```
    
Any idea on how I can achieve my goal?
    
Thank you

**Comment by vfsfitvnm** [comment]
Hi! You van follow this message: https://github.com/vfsfitvnm/frida-il2cpp-bridge/issues/50#issuecomment-971356897

**Reply by peiga** [reply]
The issue you linked has been deleted.

**Reply by vfsfitvnm** [reply]
https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/145
https://github.com/vfsfitvnm/frida-il2cpp-bridge/issues/340

**Comment by RealIndica** [comment]
> Hi! You van follow this message: [#50 (comment)](https://github.com/vfsfitvnm/frida-il2cpp-bridge/issues/50#issuecomment-971356897)

Worked flawlessly, thank you!
